packages/server/dataloader/customLoaderMakers.ts

```diff
@@ -125,31 +132,67 @@ export const userTasks = (parent: RethinkDataLoader) => {
   return new DataLoader<UserTasksKey, Task[], string>(
     async (keys) => {
       const r = await getRethink()
-      const userIds = keys.map(({userId}) => userId)
-      const teamIds = Array.from(new Set(keys.flatMap(({teamIds}) => teamIds)))
-      const taskLoader = parent.get('tasks')
-      const allUsersTasks = (await r
-        .table('Task')
-        .getAll(r.args(userIds), {index: 'userId'})
-        .filter((task) => r(teamIds).contains(task('teamId')))
-        .filter((task) =>
-          task('tags')
-            .contains('archived')
-            .not()
+      const uniqKeys = keys.reduce((dedupedKeys, currKey) => {
+        const duplicateKey = dedupedKeys.find(
+          (key) => serializeUserTasksKey(key) === serializeUserTasksKey(currKey)
         )
-        .run()) as Task[]
-      allUsersTasks.forEach((task) => {
+        if (!duplicateKey) {
+          return dedupedKeys.concat([currKey])
+        } else {
+          return dedupedKeys
+        }
+      }, [] as UserTasksKey[])
+      const taskLoader = parent.get('tasks')
+
+      const entryArray = await Promise.all(
+        uniqKeys.map(async (key) => {
+          const {first, after, userId, teamIds, archived} = key
+          const dbAfter = after ? new Date(after) : r.maxval
+          const sortedTeamIds = teamIds.sort()
+
+          return {
+            key: serializeUserTasksKey(key),
+            data: await r
+              .table('Task')
+              .between(
+                [sortedTeamIds[0], r.minval],
```

Comment:
i don't think this does what you want it to. you're only providing the first & last value in an array, so it will grab every ID that is lexicographically in between (e.g. from `aa` to `bb` it'll grab `ab, ac, ad...ba, bb` that means it could grab teamIds that the user doesn't have permission to view.

consider `.getAll(userId, {index: 'userId'}).filter(row => r(teamIds).contains(row('teamId'))`
or the inverse `.getAll(r.args(teamIds), {index: 'teamId}).filter({userId})`

in the future, we may want to add a secondary compound index on the Task object that includes the archive tag. let me know your thoughts https://rethinkdb.com/docs/secondary-indexes/javascript/#compound-indexes

Reply:
Oh you are right! I misunderstood one of your previous comments:  
> for each deduped key, make a paginated query to the DB using `between`. see [https://github.com/ParabolInc/parabol/blob/3935ArchivedCheckbox/packages/server/graphql/queries/archivedTasksCount.ts#L29](https://github.com/ParabolInc/parabol/blob/3935ArchivedCheckbox/packages/server/graphql/queries/archivedTasksCount.ts#L29)

I think adding a compound index on archive tag will be a good idea for next iteration.

Reply:
Sorry maybe I still didn't get what you are really asking here. If we query by `teamIds` and then filter by `userId` or the other way around, how are we doing paginated query? Where is the argument `after` used?

Reply:
Oh we should use that to filter by `createdAt` or `updatedAt`, right?